plugins/GSdx/GSUtil.cpp

```diff
@@ -396,6 +396,39 @@ D3D_FEATURE_LEVEL GSUtil::CheckDirect3D11Level(IDXGIAdapter *adapter, D3D_DRIVER
 	return SUCCEEDED(hr) ? level : (D3D_FEATURE_LEVEL)0;
 }
 
+GSRendererType GSUtil::GetBestRenderer()
+{
+	CComPtr<IDXGIAdapter1> adapter;
+	DXGI_ADAPTER_DESC1 desc;
+	CComPtr<IDXGIFactory1> DXGI;
+
+	if (GSUtil::CheckDXGI())
+	{
+		CreateDXGIFactory1(__uuidof(IDXGIFactory1), (void**)&DXGI);
+	}
+
+	if(GSUtil::CheckD3D11())
+	{
+		DXGI->EnumAdapters1(NULL, &adapter);
```

Comment:
> If this fails, won't the next statement dereference an null pointer and crash?

Nice catch :+1: 

> Oh, and why NULL instead of 0 since the parameter type is UINT? When I see NULL, I assume it's a null pointer.

``NULL`` is just an ``int`` , why shouldn't it be used for a ``UINT`` type parameter ? I think you're confusing it with ``nullptr``.